library/ssl_msg.c

```diff
@@ -3314,11 +3314,11 @@ int mbedtls_ssl_prepare_handshake_record(mbedtls_ssl_context *ssl)
         int ret;
         const size_t hs_remain = ssl->in_hslen - ssl->in_hsfraglen;
         MBEDTLS_SSL_DEBUG_MSG(3,
-                              ("handshake fragment: %" MBEDTLS_PRINTF_SIZET " .. %"
+                              ("handshake fragment: %u .. %"
                                MBEDTLS_PRINTF_SIZET " of %"
                                MBEDTLS_PRINTF_SIZET " msglen %" MBEDTLS_PRINTF_SIZET,
                                ssl->in_hsfraglen,
-                               ssl->in_hsfraglen +
+                               (size_t) ssl->in_hsfraglen +
```

Comment:
i think there may be a problem here if sizeof(size_t) != sizeof(unsigned). even if there isn't, to make it completely obvious i think it'd be better to leave `MBEDTLS_PRINTF_SIZET` format specifier and cast the entire arg expression   to `size_t`.

Reply:
Isn't that what I did? The CI isn't complaining as of da8f72e7f9c0ab0fe49b308aedf6fa10aff6e119, whereas it did before that commit.

Reply:
ok, my apologies, i misread. i thought the %u change of the format string applied to the wrong argument.
indeed you're right, please ignore.